papis/commands/tag.py

```diff
@@ -0,0 +1,193 @@
+"""
+This command allows users to add and remove tags to documents in their library.
+
+Examples
+^^^^^^^^
+
+- To add a tag use the `--new` (or `-n`) option:
+
+    .. code:: sh
+
+        papis tag --new TAG1 --new TAG2 QUERY
+
+  This will add TAG1 and TAG2 to a document matched by QUERY. You can repeat
+  `--new` as many times as you need. The query is any query supported by papis.
+  If the query matches more than one document, Papis' picker will be started to
+  let you pick the document from those matched (just as with Papis' other
+  commands).
+
+  Tags are only added if they do not already exist, which is to say that the
+  same tag cannot exist more than once in a given document.
+
+- To remove a tag use the `--remove` (or `-r`) option:
+
+    .. code:: sh
+
+        papis tag --remove TAG1 QUERY
+
+  This removes TAG1 from the document. If it didn't exist before, nothing
+  happens. You can use `--remove` as many times as you like to remove multiple
+  tags.
+
+- Use `--drop` (or `-d`) to remove all tags:
+
+    .. code:: sh
+
+        papis tag --drop --new TAG1 QUERY
+
+  This removes all tags from the document and then adds TAG1. You can of course
+  also use `--drop` by itself to simply remove all tags without adding new
+  ones.
+
+- Use `--all` (or `-a`) with any of the other options to apply the tagging
+  operation to all matching documents:
+
+    .. code:: sh
+
+        papis tag --all --new TAG1 QUERY
+
+  This adds TAG1 to all documents matching the QUERY rather than opening the
+  picker to let you choose one.
+
+"""
+from typing import List, Optional, Tuple, Any
+
+import click
+
+import papis.cli
+import papis.commands
+import papis.importer
+import papis.commands.update
+import papis.commands.doctor
+import papis.config
+import papis.document
+import papis.strings
+import papis.tui.utils
+
+logger = papis.logging.get_logger(__name__)
+
+
+@click.command("tag")
+@click.help_option("--help", "-h")
+@papis.cli.git_option()
+@papis.cli.bool_flag(
+    "-d",
+    "--drop",
+    help="Drop all tags.",
```

Comment:
```suggestion
    help="Drop all tags",
```
Very nitpicky, but these help messages don't usually have a full stop at the end?